### PR TITLE
CIRC-256 Send request-related notices

### PR DESCRIPTION
**Pull request by OleksiiMaksymov**
**Purpose**
Send request-related notices (item status is changed to "awaiting pickup")

**Link**
[CIRC-256](https://issues.folio.org/browse/CIRC-256)

### CODE REVIEW

File: src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java [hunk 1]
```diff
@@ -127,4 +135,37 @@
...
+          userRepository.getUserByBarcode(barcode)
+            .thenAccept(r -> r.next(sendPatronNotice(records.getItem())));
+        }
+      );
+    return succeeded(records);
+  }
+
+  private Function<User, Result<User>> sendPatronNotice(Item item) {
+    return user -> {
+      if (AWAITING_PICKUP == item.getStatus()) {
```
Comment (kostyantyn-kh) [inline src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java]: Please add method _isAwaitingPickup_ to Item and use it here

File: src/main/java/org/folio/circulation/domain/Request.java [hunk 2]
```diff
@@ -197,6 +197,11 @@ public User getRequester() {
     return requester;
   }
 
+  public String getRequesterBarcode(){
+    JsonObject userData = asJson().getJsonObject("requester", new JsonObject());
```
Comment (kostyantyn-kh) [inline src/main/java/org/folio/circulation/domain/Request.java]: please rework this method to work with _representation_ object directly

Comment (OleksiiMaksymov) [reply src/main/java/org/folio/circulation/domain/Request.java]: fix

File: src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java (hunk 1 above)
Comment (OleksiiMaksymov) [reply src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java]: done

**Pull request merged**